Gems/Meshlets/Code/Source/Meshlets/SharedBuffer.h

```diff
@@ -0,0 +1,156 @@
+            //! The format used for the buffer
+            //! Should be Unknown for structured buffers, or R32 for raw buffers.
+            RHI::Format m_elementFormat;
+            //! The size in bytes of each element in the stream
+            RHI::BufferBindFlags m_bindFlags;
+
+            uint32_t m_elementSize;
+            //! Amount of elements required to create the buffer
+            uint32_t m_elementCount;
+            //! The name used for the buffer view - used for debug and tracking
+            Name m_bufferName;
+            //! The name used by the shader Srg in the GPU for this shader parameter 
+            Name m_paramNameInSrg;
+            //! The assigned SRG slot in the CPU / GPU for this shader resource
+            uint32_t m_resourceShaderIndex;
+            //! If using a buffer view within a shared buffer, this represents
+            //! the view offset from the shared buffer origin in bytes.
+            uint32_t m_viewOffsetInBytes;
+
+            uint8_t* m_bufferData = nullptr;
+
+            SrgBufferDescriptor() = default;
+            SrgBufferDescriptor(
+                RPI::CommonBufferPoolType poolType,
+                RHI::Format elementFormat,
+                RHI::BufferBindFlags m_bindFlags,
+                uint32_t elementSize,
+                uint32_t elementCount,
+                Name bufferName,
+                Name paramNameInSrg,
+                uint32_t resourceShaderIndex,
+                uint32_t viewOffsetInBytes,
+                uint8_t* bufferData = nullptr
+            ) : m_poolType(poolType),
+                m_elementFormat(elementFormat), m_bindFlags(m_bindFlags),
+                m_elementSize(elementSize), m_elementCount(elementCount),
+                m_bufferName(bufferName), m_paramNameInSrg(paramNameInSrg),
+                m_resourceShaderIndex(resourceShaderIndex), m_viewOffsetInBytes(viewOffsetInBytes),
+                m_bufferData(bufferData)
+            {};
+        };
+
+        //!=====================================================================
+        //! This class represents a single RPI::Buffer used to allocate sub-buffers from the
+        //!  existing buffer that can then be used per draw.   In a way, this buffer is used as a memory
+        //!  pool from which sub-buffers are being created.
+        //! This is very useful when we want to synchronize the use of these buffers via barriers 
+        //!  so we declare and pass the entire buffer between passes and therefore we are creating
+        //!  a dependency and barrier for this single buffer, yet as a result all sub-buffers are
+        //!  now getting synced between passes.
+        //! 
+        // Adopted from SkinnedMeshOutputStreamManager in order to make it more generic and context free usage
+        //!=====================================================================
+        class SharedBuffer
```

Comment:
this is a very good idea, but comparing both of them and the one you created for the skinning - each has its own unique requirements that are not always aligned.
I strongly support what you suggested but it will require us to come up with better creation of the pools (not via the common pool), and better generic interface that will allow support of a map of these shared buffers.
Bottom line - yes to your suggestion to generalize this shared buffer concept, I also suggested it in the past and started testing it, however, it requires more steps as I mentioned.